### DISCUSSION TITLE
User driven theming

### DISCUSSION THREAD

**Discussion by kijv** [fuma-nama/fumadocs#848] [Ideas]
A few months ago I was experimenting with my own docs tool that made the theme extensible, by default, already given the tools to build a good decumentarion experiment.

I think **creating a system that allows the user to either go with the default Fumadocs theme, or providing various arguments to create their own UI** for the documentation while still relying on the backbone of Fumadocs (whether by a “preset” or user-provided values)

I have been busy recently and have been unable to research and iterate on this idea personally, but I wanted to suggest this here because Fumadocs already has a well-built infrastructure for most (if not all) functionality expected from a docs framework.

This will ideally make the Fumadocs UI package a theme that is used by default if it’s installed. I personally see this the feature stopping me from fully switching to Fumadocs, because of the rigid UI freedom.

**I’m happy to work on this more (and possibly add it to fumadocs)** if this is something you see will be beneficial to the users.

I hope you consider this idea, thanks!

**Comment by fuma-nama** [comment]
why don’t use CSS variables? Fumadocs UI was based on Shadcn UI

**Reply by kijv** [reply]
I'm not referring to styling alone. This idea is to control every part of the layout besides the actual documentation content from the MDX.

**Reply by Gitstar-OC** [reply]
Well you should develop that, it seems great until people uses it. Although I would love to try that!

**Reply by fuma-nama** [reply]
Then you can make your own theme for Fumadocs Core, I'm not planning to maintain the feature/codebase even if you submitted a PR. This doesn't align with the goals of Fumadocs.

And some degrees of customisation can be done with CSS and client components. You can also read themes from server components and pass it to Fumadocs UI.

**Reply by kijv** [reply]
Ok, I see where you come from in terms of maintenance, especially since this doesn't with you goals for the project.

**Reply by kijv** [reply]
> Well you should develop that, it seems great until people uses it. Although I would love to try that!

Thanks! Hopefully I can get the chance to work on this idea more and make a production grade version eventually
